### PR TITLE
Risk calculation failed card - ui (EXPOSUREAPP-3051)

### PR DESCRIPTION
**Pull request by SamuraiKek**
### Description

This PR introduces a new risk card for when the comparison of the random id's with the server fails. 

We are also trying to separate the risk cards in different views based on the risk status, starting with this one. Up until now, we've only had one view with a data class responsible for all the states of the risk card which is growing quite clunky and hard to follow.  (TracingCardState)

This PR is open as draft for now as there are other views that need to be implemented for the rest of the risk card states and with my availability reducing, other people could contribute to it as well.

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/riskcards/RiskCardNoInternetStateProvider.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import javax.inject.Inject
+
+@Reusable
+class RiskCardNoInternetStateProvider @Inject constructor(
+    tracingRepository: TracingRepository
+) {
+
+    val state: Flow<RiskCardNoInternet> = combine(
+        RiskLevelRepository.riskLevelScoreLastSuccessfulCalculated.onEach {
+            Timber.v("riskLevelScoreLastSuccessfulCalculated: $it")
```
Comment (ritsam) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/riskcards/RiskCardNoInternetStateProvider.kt:21]: do you think we can use natural words here instead ofusing camel case variables?
More of a question than a suggestion.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/home/HomeFragment.kt [hunk 2]
```diff
@@ -91,7 +102,9 @@ class HomeFragment : Fragment(R.layout.fragment_home), AutoInject {
         }
 
         vm.showLoweredRiskLevelDialog.observe2(this) {
-            if (it) { showRiskLevelLoweredDialog() }
+            if (it) {
+                showRiskLevelLoweredDialog()
+            }
```
Comment (ritsam) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/home/HomeFragment.kt:107]:
```suggestion
 if (it) { showRiskLevelLoweredDialog() }
```

Probably your lint settings changed it. but makes sense to have it in single line

**Review by ritsam — COMMENTED** [review]
onViewCreated() method can have less lines. detekt issue might crop up I feel.

You think moving the _observe2_ declarations to another funtion make sense?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/home/HomeFragment.kt (hunk 2 above)
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/home/HomeFragment.kt:107]: You're right, it was the linter doing it's stuff. I agree, it should be a one liner.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/riskcards/RiskCardNoInternetStateProvider.kt (hunk 1 above)
Comment (SamuraiKek) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/riskcards/RiskCardNoInternetStateProvider.kt:21]: It might be a good idea. I just followed the other state providers as an example for this (SubmissionCardsStateProvider) and they all have the Timber messages like this. Right now it looks like the Timber logs differ from person to person and  I don't really think we have some guidelines for them.

**Comment by SamuraiKek** [comment]
Closing this draft PR as it's no longer relevant.
